Tasks/Common/nuget-task-common/NuGetToolRunner.ts

```diff
@@ -91,29 +91,53 @@ export function createNuGetToolRunner(nuGetExePath: string, settings: NuGetEnvir
     return runner;
 }
 
-export function locateTool(tool: string, userPath?: string, optional?: boolean) {
-    tl.debug("looking for tool " + tool);
-    if (userPath) {
-        tl.debug("using user-supplied path " + userPath)
-        tl.checkPath(userPath, 'nuget');
-        return userPath;
-    }
+export interface LocateOptions {
+    optional?: boolean;
+    userPath?: string;
+    fallbackToSystemPath?: boolean;
+}
 
+export function locateTool(tool: string | string[], opts?: LocateOptions) {
+    let toolVariants: string[] = typeof tool === 'string' ? [tool] : tool;
     let searchPath = ["externals/nuget", "agent/Worker/Tools/NuGetCredentialProvider", "agent/Worker/Tools"];
-
     let agentRoot = tl.getVariable("Agent.HomeDirectory");
-    for (let possibleLocation of searchPath) {
-        let fullPath = path.join(agentRoot, possibleLocation, tool);
-        tl.debug("checking " + fullPath);
-        if (tl.exist(fullPath)) {
-            return fullPath;
-        }
+
+    opts = opts || {};
+
+    tl.debug(`looking for tool ${tool}`)
+
+    if (opts.userPath) {
+        tl.debug(`using user-supplied path ${opts.userPath}`)
+        tl.checkPath(opts.userPath, toolVariants[0]);
```

Comment:
Did you mean intend to make the assumption that `toolVariants[0]` will always work? Consider adding a comment explaining why that (seemingly incorrect) assumption is safe.